spring-integration-core/src/main/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiator.java

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.leader;
```

Comment:
M-m-m, do I understand correctly that you have been forced to place this class here, in this `support` sub-package, because we have a package tangle with the top-level `leader` where `SmartLifecycleRoleController` in `support` uses `leader.event`?

I don't know how to build properly dependency matrix in the IDEA, but that really feels like `SmartLifecycleRoleController` should move to the top-level `leader` package to bring `support` one closer to dependency-free of all others.